### PR TITLE
Wait for query cancellation for JDBC and CLI

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: presto-client/src/main/java/io/prestosql/client/StatementClientV1.java [hunk 1]
```diff
@@ -493,7 +493,12 @@ private void httpDelete(URI uri)
         Request request = prepareRequest(HttpUrl.get(uri))
                 .delete()
                 .build();
-        httpClient.newCall(request).enqueue(new NullCallback());
+        try {
+            httpClient.newCall(request).execute().close();
+        }
+        catch (IOException ignored) {
+            // callers expect this method not to throw
```
Comment (kokosing) [inline presto-client/src/main/java/io/prestosql/client/StatementClientV1.java]: please log it as debug

File: presto-client/src/main/java/io/prestosql/client/StatementClientV1.java [hunk 2]
```diff
@@ -493,7 +493,12 @@ private void httpDelete(URI uri)
         Request request = prepareRequest(HttpUrl.get(uri))
                 .delete()
                 .build();
-        httpClient.newCall(request).enqueue(new NullCallback());
+        try {
+            httpClient.newCall(request).execute().close();
```
Comment (kokosing) [inline presto-client/src/main/java/io/prestosql/client/StatementClientV1.java]:
```
httpClient.newCall(request)
    .execute()
    .close();
```

File: presto-client/src/main/java/io/prestosql/client/StatementClientV1.java (hunk 1 above)
Comment (electrum) [reply presto-client/src/main/java/io/prestosql/client/StatementClientV1.java]: We don't have any logging in `presto-client` so I'm hesitant to add it just for this, as that could cause problems for the JDBC driver. The existing code does it async and ignores the result, so it isn't any worse. Also, if we cared about the response, we'd need to check that it was a successful HTTP request (it only throws if there is an I/O error, not due to HTTP 500 etc.)

Comment (findepi) [reply presto-client/src/main/java/io/prestosql/client/StatementClientV1.java]: Does this mean we may fail to cancel and ignore this?

**Pull request merged**